sdk/servicebus/Azure.Messaging.ServiceBus/src/Transports/Amqp/ServiceBusMessageExtensions.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Azure.Messaging.ServiceBus.Transports.Amqp
+{
+    /// <summary>
+    /// Extension methods for <see cref="ServiceBusMessage"/>.
+    /// </summary>
+    public static class ServiceBusMessageExtensions
+    {
+        /// <summary>
+        /// Checks if the message contains an AMQP body.
+        /// </summary>
+        /// <param name="message">The message to use.</param>
+        /// <returns>True if the message contains an AMQP body. Otherwise, false.</returns>
+        public static bool HasAmqpBody(this ServiceBusMessage message) => message.TransportBody is AmqpTransportBody;
+
+        /// <summary>
+        /// Gets the body type from an AMQP body.
+        /// </summary>
+        /// <param name="message">The message to use.</param>
+        /// <returns>If the message contains an AMQP body, returns the body type. Otherwise, <see cref="AmqpBodyType.Unspecified"/>.</returns>
+        public static AmqpBodyType GetAmqpBodyType(this ServiceBusMessage message) =>
+            message.TransportBody is AmqpTransportBody body ? body.BodyType : AmqpBodyType.Unspecified;
```

Comment:
Would rather wait to add handling for non-AMQP body until we actually need it. Otherwise we double the number of code paths.

Reply:
So, you want me to just assume? Like, cast all of them? I'm fine with that.

Reply:
Ah right, BodyType is not part of the abstraction so we still need to cast. I don't really have a strong opinion here. @jsquire ?

Reply:
Personally, I still prefer doing the cast as we know it is the correct type right now. It also avoids adding untested branches.

Reply:
I've changed it to a cast. Note: There was an advantage to the `is` check... if for any reason `TransportBody` was `null`, all the code will still function properly. With this cast, we lose this guarantee.